tfhe/src/strings/assert_functions/test_common.rs

```diff
@@ -0,0 +1,367 @@
+        let dec_is_some = self.ck.decrypt_bool(&is_some);
+        if !dec_is_some {
+            // When it's None, the FheString returned is the original str
+            assert_eq!(dec_result, str);
+        }
+
+        let dec = dec_is_some.then_some(dec_result.as_str());
+
+        println!("\n\x1b[1mStrip_prefix:\x1b[0m");
+        result_message_pat(str, pat, expected, dec, end.duration_since(start));
+
+        assert_eq!(dec, expected);
+
+        let start = Instant::now();
+        let (result, is_some) = self.sk.strip_prefix(&enc_str, &clear_pat);
+        let end = Instant::now();
+
+        let dec_result = self.ck.decrypt_ascii(&result);
+        let dec_is_some = self.ck.decrypt_bool(&is_some);
+        if !dec_is_some {
+            // When it's None, the FheString returned is the original str
+            assert_eq!(dec_result, str);
+        }
+
+        let dec = dec_is_some.then_some(dec_result.as_str());
+
+        println!("\n\x1b[1mStrip_prefix:\x1b[0m");
+        result_message_clear_pat(str, pat, expected, dec, end.duration_since(start));
+
+        assert_eq!(dec, expected);
+    }
+
+    pub fn assert_strip_suffix(
+        &self,
+        str: &str,
+        str_pad: Option<u32>,
+        pat: &str,
+        pat_pad: Option<u32>,
+    ) {
+        let expected = str.strip_suffix(pat);
+
+        let enc_str = self.encrypt_string(str, str_pad);
+        let enc_pat = GenericPattern::Enc(self.encrypt_string(pat, pat_pad));
+        let clear_pat = GenericPattern::Clear(ClearString::new(pat.to_string()));
+
+        let start = Instant::now();
+        let (result, is_some) = self.sk.strip_suffix(&enc_str, &enc_pat);
+        let end = Instant::now();
+
+        let dec_result = self.ck.decrypt_ascii(&result);
+        let dec_is_some = self.ck.decrypt_bool(&is_some);
+        if !dec_is_some {
+            // When it's None, the FheString returned is the original str
+            assert_eq!(dec_result, str);
+        }
+
+        let dec = dec_is_some.then_some(dec_result.as_str());
+
+        println!("\n\x1b[1mStrip_suffix:\x1b[0m");
+        result_message_pat(str, pat, expected, dec, end.duration_since(start));
+
+        assert_eq!(dec, expected);
+
+        let start = Instant::now();
+        let (result, is_some) = self.sk.strip_suffix(&enc_str, &clear_pat);
+        let end = Instant::now();
+
+        let dec_result = self.ck.decrypt_ascii(&result);
+        let dec_is_some = self.ck.decrypt_bool(&is_some);
+        if !dec_is_some {
+            // When it's None, the FheString returned is the original str
+            assert_eq!(dec_result, str);
+        }
+
+        let dec = dec_is_some.then_some(dec_result.as_str());
+
+        println!("\n\x1b[1mStrip_suffix:\x1b[0m");
+        result_message_clear_pat(str, pat, expected, dec, end.duration_since(start));
+
+        assert_eq!(dec, expected);
+    }
+
+    pub fn assert_comp(&self, str: &str, str_pad: Option<u32>, rhs: &str, rhs_pad: Option<u32>) {
```

Comment:
Seems like this is missing >=, <=, >, < with clear rhs,

And optionally, maybe we can factorize code by having a function `fn assert_comp_impl(&self, lhs_clear: &str, enc_lhs: FheString rhs_clear: &str, rhs: :  GenericPattern)`

Reply:
These functions don't support a clear rhs

Reply:
I suspected that, technically it could be supported, but that's ok for now